pkg/daemon/update.go

```diff
@@ -83,19 +99,65 @@ func (dn *Daemon) performPostConfigChangeAction(postConfigChangeActions []string
 		if dn.nodeWriter != nil {
 			dn.nodeWriter.Eventf(corev1.EventTypeNormal, "SkipReboot", "Config changes do not require reboot.")
 		}
+		err := upgrademonitor.GenerateAndApplyMachineConfigNodes(
+			&upgrademonitor.Condition{State: mcfgalphav1.MachineConfigNodeUpdatePostActionComplete, Reason: "None", Message: fmt.Sprintf("Changes do not require a reboot")},
+			&upgrademonitor.Condition{State: mcfgalphav1.MachineConfigNodeUpdateRebooted, Reason: fmt.Sprintf("%s%s", string(mcfgalphav1.MachineConfigNodeUpdatePostActionComplete), "None"), Message: "No reboot or reaload required"},
+			metav1.ConditionTrue,
+			metav1.ConditionTrue,
+			dn.node,
+			dn.mcfgClient,
+			dn.featureGatesAccessor,
+		)
+		if err != nil {
+			klog.Errorf("Error making MCN: %w", err)
+		}
 		logSystem("Node has Desired Config %s, skipping reboot", configName)
 	}
 
 	if ctrlcommon.InSlice(postConfigChangeActionReloadCrio, postConfigChangeActions) {
 		serviceName := "crio"
-
+		//annos := map[string]string{
+		//constants.MachineConfigDaemonStateAnnotationKey: constants.MachineConfigDaemonStateWorkPostAction,
+		//constants.MachineConfigDaemonPhaseAnnotationKey:  string(mcfgalphav1.r),
+		//constants.MachineConfigDaemonReasonAnnotationKey: "Reloading CRIO, upgrade does not require reboot.",
+		//}
+		//_, err := dn.nodeWriter.SetAnnotations(annos)
+		//if err != nil {
+		//	return err
+		//}
+		err := upgrademonitor.GenerateAndApplyMachineConfigNodes(
+			&upgrademonitor.Condition{State: mcfgalphav1.MachineConfigNodeUpdatePostActionComplete, Reason: string(mcfgalphav1.MachineConfigNodeUpdateReloaded), Message: "Node will reload CRIO"},
+			&upgrademonitor.Condition{State: mcfgalphav1.MachineConfigNodeUpdateReloaded, Reason: fmt.Sprintf("%s%s", string(mcfgalphav1.MachineConfigNodeUpdatePostActionComplete), string(mcfgalphav1.MachineConfigNodeUpdateReloaded)), Message: "Upgrade requires a CRIO reload. Currently doing this as the post update action."},
+			metav1.ConditionUnknown,
+			metav1.ConditionUnknown,
+			dn.node,
+			dn.mcfgClient,
+			dn.featureGatesAccessor,
+		)
+		if err != nil {
+			klog.Errorf("Error making MCN: %w", err)
+		}
+		//dn.EmitUpgradeEvent(dn.stateControllerPod, dn.UpgradeAnnotations(mcfgv1.MachineConfigNodeUpdatePostActionComplete), corev1.EventTypeNormal, "RealoadingCRIO", fmt.Sprintf("Realoding CRIO on node %s as part of upgrade", dn.nodeName()))
 		if err := reloadService(serviceName); err != nil {
 			if dn.nodeWriter != nil {
 				dn.nodeWriter.Eventf(corev1.EventTypeWarning, "FailedServiceReload", fmt.Sprintf("Reloading %s service failed. Error: %v", serviceName, err))
 			}
 			return fmt.Errorf("could not apply update: reloading %s configuration failed. Error: %w", serviceName, err)
 		}
 
+		err = upgrademonitor.GenerateAndApplyMachineConfigNodes(
+			&upgrademonitor.Condition{State: mcfgalphav1.MachineConfigNodeUpdatePostActionComplete, Reason: string(mcfgalphav1.MachineConfigNodeUpdateReloaded), Message: "Node has reloaded CRIO"},
```

Comment:
The service reload should also almost be instant, should we consider collapsing this and the above state?

Reply:
sure, we can skip to true.